library/ssl_msg.c

```diff
@@ -4036,6 +4140,14 @@ int mbedtls_ssl_read_record( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
                 {
                     ret = ssl_get_next_record( ssl );
+
+#if defined(MBEDTLS_SSL_HS_DEFRAG_CLT)
+                    if( ( ret == 0 ) &&
+                        ( mbedtls_ssl_hs_defrag_enabled( ssl ) ) &&
+                        ( ssl->in_msgtype == MBEDTLS_SSL_MSG_HANDSHAKE ) )
+                        ret = ssl_defrag_accumulate( ssl );
+#endif /* MBEDTLS_SSL_HS_DEFRAG_CLT */
+
```

Comment:
This is a bit hard to read. Some suggestions for improvement:
1) Keep the check for `ssl_get_next_record();`, to maintain the usual flow of call,check return value, continue.
2) Minor: Logically, it seems more natural to check the record type first before calling `mbedtls_ssl_hs_defrag_enabled()`.
3) Call, check, continue in the body of the `if` branch.
```suggestion
                    ret = ssl_get_next_record( ssl );
                    if( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING )
                        continue;
                    if( ret != 0 )
                    {
                        MBEDTLS_SSL_DEBUG_RET( 1, ( "ssl_get_next_record" ), ret );
                        return( ret );
                    }

#if defined(MBEDTLS_SSL_HS_DEFRAG_CLT)
                    if( ssl->in_msgtype == MBEDTLS_SSL_MSG_HANDSHAKE &&
                        mbedtls_ssl_hs_defrag_enabled( ssl ) )
                    {
                        ret = ssl_defrag_accumulate( ssl );
                        if( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING )
                            continue;
                        if( ret != 0 )
                            return( ret );
                    }
#endif /* MBEDTLS_SSL_HS_DEFRAG_CLT */
```

Reply:
Applied the suggested changes.